### PR TITLE
[MM-15268] Add a config.json setting for minimum username length

### PR DESCRIPTION
**Pull request by carmo-evan**
#### Summary

[MM-15268] Add a config.json setting for minimum username length

#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/10702

#### Related Pull Requests
- Has server changes (please link here)

[MM-15268]: https://mattermost.atlassian.net/browse/MM-15268

### CODE REVIEW

File: i18n/es.json [hunk 1]
```diff
@@ -515,6 +515,7 @@
   "admin.field_names.maxChannelsPerTeam": "Máximo número de Canales por Equipo",
   "admin.field_names.maxNotificationsPerChannel": "Cantidad máxima de Notificaciones por Canal",
   "admin.field_names.maxUsersPerTeam": "Máximo de usuarios por equipo",
+  "admin.field_names.minimumUsernameLength": "Longitud mínima del nombre de usuario",
```
Comment (jespino) [inline i18n/es.json]: We only include the `en.json` string, all the other strings are managed through our translations server (https://translate.mattermost.com/).

File: components/user_settings/advanced/index.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ function makeMapStateToProps() {
         return {
             advancedSettingsCategory: getAdvancedSettingsCategory(state, Preferences.CATEGORY_ADVANCED_SETTINGS),
             sendOnCtrlEnter: get(state, Preferences.CATEGORY_ADVANCED_SETTINGS, 'send_on_ctrl_enter', 'false'),
-            codeBlockOnCtrlEnter: get(state, Preferences.CATEGORY_ADVANCED_SETTINGS, 'code_block_ctrl_enter', 'true'),
+            codeBlockOnCtrlEnter: getBool(state, Preferences.CATEGORY_ADVANCED_SETTINGS, 'code_block_ctrl_enter'),
```
Comment (jespino) [inline components/user_settings/advanced/index.js]: This looks like a completely unrelated change

File: components/admin_console/admin_definition.jsx [hunk 3]
```diff
@@ -4128,3 +4138,4 @@ t('admin.field_names.postEditTimeLimit');
 t('admin.field_names.restrictCreationToDomains');
 t('admin.field_names.restrictDirectMessage');
 t('admin.field_names.teammateNameDisplay');
+t('admin.field_names.minimumUsernameLength');
```
Comment (jespino) [inline components/admin_console/admin_definition.jsx]: I think this here is not needed.

File: i18n/es.json (hunk 1 above)
Comment (carmo-evan) [reply i18n/es.json]: I was getting failures in this test https://github.com/mattermost/mattermost-webapp/blob/master/utils/admin_console_index.test.jsx that only cleared after I added the Spanish translation.

File: components/user_settings/advanced/index.js (hunk 2 above)
Comment (carmo-evan) [reply components/user_settings/advanced/index.js]: I'll fix.

File: components/admin_console/admin_definition.jsx (hunk 3 above)
Comment (carmo-evan) [reply components/admin_console/admin_definition.jsx]: I'll test it without it.

File: i18n/es.json (hunk 1 above)
Comment (jespino) [reply i18n/es.json]: Yes, that makes sense, because the test is doing the check against the translations, I didn't think about this exact case. Not a super cool solution, but we could change the search text used for the tests to not involve this change.

**Comment by hanzei** [comment]
Closing this one due to inactivity. @RajatVaryani will now work on this topic.
